paddle/fluid/framework/ir/mkldnn_correct_test_phase_pass.cc

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "paddle/fluid/framework/ir/mkldnn_correct_test_phase_pass.h"
+#include <string>
+#include <utility>
+
+namespace paddle {
+namespace framework {
+namespace ir {
+
+std::unique_ptr<ir::Graph> MKLDNNCorrectTestPhasePass::ApplyImpl(
+    std::unique_ptr<ir::Graph> graph) const {
+  VLOG(3) << "Sets MKL-DNN is_test attrbiute.";
+  std::array<std::string, 31> op_list = {
+      "pool2d",      "sigmoid",      "logsigmoid",
+      "softshrink",  "exp",          "brelu",
+      "pow",         "leaky_relu",   "stanh",
+      "relu",        "tanh",         "tanh_shrink",
+      "sqrt",        "abs",          "ceil",
+      "elu",         "floor",        "cos",
+      "sin",         "round",        "reciprocal",
+      "hard_shrink", "hard_sigmoid", "relu6",
+      "soft_relu",   "swish",        "thresholded_relu",
+      "log",         "square",       "softplus",
+      "softsign"};
+  for (const Node* n : graph->Nodes()) {
+    if (n->IsOp()) {
+      auto* op = n->Op();
+      if (op->HasAttr("use_mkldnn") && op->GetAttr("use_mkldnn")) {
```

Comment:
why need `op->GetAttr("use_mkldnn")`?

Reply:
It is my mistake. Non MKLDNN layers could also be optimized for inference, right? 
I am not sure if `is_test` flag has the same use in all layers especially in the below 3 layers.

1. `fake_quantize_op.cc` has `AddAttr<bool>("is_test", "").SetDefault(false);`. 
2. `open_files_op.cc` has  ` AddAttr<bool>("is_test", "Used for testing data.").SetDefault(false);`. 
3. `while_op.cc`. has `AddAttr<bool>("is_test", "True if in test phase.").SetDefault(false);`

They seem to be used the same way but can you confirm?